### DISCUSSION TITLE
Filter for datetime field fails with string convertion with UTC timezone and plus sign (bug?)

### DISCUSSION THREAD

**Discussion by aemitos** [encode/django-rest-framework#7948] [Potential Issue]
### Description

We implemented a test that was using naive datetime to filter. Once we made it timezone aware using `timezone.now()` the string used on the URL became like this:

```
'/api/v1/<our_view_set_url>?created_after=2021-04-20 23:45:23.405285+00:00'
```
We can see that the offset was added with a `+`. 
That returns for us an error (400) indicating that we provided an invalid date/time option to the filter.
When I checked the `django_filters.rest_framework.backends.py`, the method:

```python
    def get_filterset(self, request, queryset, view):
        filterset_class = self.get_filterset_class(view, queryset)
        if filterset_class is None:
            return None

        kwargs = self.get_filterset_kwargs(request, queryset, view)
        return filterset_class(**kwargs)
```

The kwargs looked like this:

```python
{'data': <QueryDict: {'created_after': ['2021-04-20 23:45:23.405285 00:00']}>,
 'queryset': <QuerySet [...<series of instances>...] >,
 'request': <rest_framework.request.Request: GET '/api/v1/<our_view_set_url>?created_after=2021-04-20%2023:45:23.405285+00:00'>}
```
We can see on the `data` that the parsing removed the `+` and added space.
That doesn't happen with timezones that have a negative offset (example: "-04:00").
We didn't set the `REST_FRAMEWORK.DATETIME_FORMAT` and our view set looks like this:

```python
class OurViewSet(ReadOnlyModelViewSet):
    queryset = OurModel.objects.none()
    serializer_class = OurModelSerializer
    filter_backends = (
        filters.OrderingFilter,
        DjangoFilterBackend,
    )
    ordering_fields = ("modified", "created")
    filterset_class = OurModelFilterSet
```

As mentioned above, the same doesn't happen for timezones with a negative offset.
Is this a bug?

**Comment by Safrone** [comment]
It's not really a "bug" given plusses are supposed to be interpreted as spaces in query strings https://developers.google.com/maps/url-encoding. 

my workaround:
```
class QueryDateTimeField(serializers.DateTimeField):
  """handles datetime with a + in the url query since that is transformed into a space"""

  def to_internal_value(self, value):
    try:
      return super().to_internal_value(value)
    except rest_framework.exceptions.ValidationError:
      if isinstance(value, str) and ' ' in value:
        # rreplace
        new_value = value[::-1].replace(' ', '+', 1)[::-1]
        return super().to_internal_value(new_value)
      raise
```

**Reply by trojjer** [reply]
Agreed. Looks like this one may be down to the stdlib `parse_qsl` which is invoked by Django's `QueryDict.__init__`. https://github.com/python/cpython/blame/99b00efd5edfd5b26bf9e2a35cbfc96277fdcbb1/Lib/urllib/parse.py#L795

```name = nv[0].replace('+', ' ')```

It's amusing to be putting back what is taken away because of a merge now 13 years ago, but seems like the cleanest way.
